src/ray/gcs/redis_context.cc

```diff
@@ -24,24 +24,31 @@ void GlobalRedisCallback(void *c, void *r, void *privdata) {
   }
   int64_t callback_index = reinterpret_cast<int64_t>(privdata);
   redisReply *reply = reinterpret_cast<redisReply *>(r);
-  std::string data = "";
-  if (reply->type == REDIS_REPLY_NIL) {
-    // Respond with blank string, which triggers a failure callback for lookups.
-  } else if (reply->type == REDIS_REPLY_STRING) {
-    data = std::string(reply->str, reply->len);
-  } else if (reply->type == REDIS_REPLY_ARRAY) {
-    reply = reply->element[reply->elements - 1];
-    data = std::string(reply->str, reply->len);
-  } else if (reply->type == REDIS_REPLY_STATUS) {
-  } else if (reply->type == REDIS_REPLY_ERROR) {
+  std::vector<std::string> data;
```

Comment:
It seems `data` always has size 1 throughout this PR.  If that's true, what's the reason of changing the iface to vector (by default I think it allocates a few more slots than the absolute minimum, which is 1; this might be wrong though)?

Reply:
Eventually we want to transition to an interface where an operation appends a value instead of just overwriting it, which isn't flexible enough to support the object table, for instance. One thing that could work is if we use the `std::vector` interface internally, but the interface for the table caller can still take in just a single `data`. Would you prefer that we do that in this PR?